src/NuGet.Core/NuGet.Packaging/Licenses/NuGetLicenseOperator.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
+namespace NuGet.Packaging
+{
+    /// <summary>
+    /// A NuGetLicenseOperator. The operator options are: WITH or Logical operator, AND and OR.
+    /// This is an abstract class so based on the NuGetLicenseOperatorType, it can be either a <see cref="WithOperator"/> or a <see cref="NuGetLicenseLogicalOperator"/>.
+    /// </summary>
+    public abstract class NuGetLicenseOperator : NuGetLicenseExpression
+    {
+        /// <summary>
+        /// The operator type.
+        /// LogicalOperator means it's AND or OR and <see cref="NuGetLicenseLogicalOperator"/>
+        /// NuGetLicenseWithOperator means it's the WITH operator and <see cref="WithOperator"/>
+        /// </summary>
+        public NuGetLicenseOperatorType OperatorType { get; protected set; }
```

Comment:
>protected [](start = 60, length = 9)

This could be made `{ get; }` if you pass the values into the base constructor right? Then we know this is a truly immutable type.